src/script/builder/tir/block_frame.h

```diff
@@ -62,15 +64,56 @@ class BlockFrameNode : public TIRFrameNode {
 };
 
 class BlockFrame : public TIRFrame {
+ public:
+  BlockFrame() = default;
+
  public:
   TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(BlockFrame, TIRFrame, BlockFrameNode);
 };
 
-BlockFrame Block_(String name);
+BlockFrame Block_(String name, bool no_realize = false);
+
+class BlockInitFrameNode : public TIRFrameNode {
+ public:
+  Optional<tvm::tir::Stmt> init;
+
+  void VisitAttrs(tvm::AttrVisitor* v) {
+    TIRFrameNode::VisitAttrs(v);
+    v->Visit("init", &init);
+  }
+
+  static constexpr const char* _type_key = "script.builder.tir.BlockInitFrame";
+  TVM_DECLARE_FINAL_OBJECT_INFO(BlockInitFrameNode, TIRFrameNode);
+
+ public:
+  void EnterWithScope() final;
+  void ExitWithScope() final;
+};
+
+class BlockInitFrame : public TIRFrame {
+ public:
+  TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(BlockInitFrame, TIRFrame, BlockInitFrameNode);
+};
+
+BlockInitFrame BlockInit();
```

Comment:
to be consistent with TVMScript syntax

```suggestion
BlockInitFrame Init();
```